task: add support for php 8.2

xaevik opened the pull request:
# Description

Adds support for PHP 8.2 for Magento >= 2.4.6 installations

- Closes #226

shochdoerfer commented:
@xaevik awesome, thanks for helping out!

shochdoerfer commented:
@xaevik the CI build fails because we sticked to Composer 1 for a long time (to be able to install the module also on older Magento instances). Now that hurts us ;( 

I think the best would be to upgrade to Composer 2 and not care about older Magento versions anymore. Since we run the builds for Magento 2.4.x, I think all those versions should be compatible with Composer 2. Or what do you think?

xaevik commented:
@shochdoerfer if you're only targeting Magento >= 2.4.x at this point, I would safely say bumping the composer version to >= 2.2 in the CI pipeline would be wise, I can update the PR if you'd like to include those changes. 

As far as I remember, stock Magento 1.x never supported composer as a first party concern, but I know the [OpenMage](https://github.com/OpenMage/magento-lts) (which is an M 1.x fork) project does.

xaevik commented:
@shochdoerfer I updated ci.yml to reflect the necessary changes:

- Set minimum composer version to 2
- Remove EOL releases of Magento
- Set minimum PHP version to 8.1

This should allow the pipeline to run properly.

xaevik commented:
Looks like a CodeSniffer dependency is missing, I'll address that shortly.

xaevik commented:
Ok, looks like some more work is required here. Unfortunately `captainhook` is giving me some difficulties on my local but I'll set some time aside to look at this deeper tomorrow.

shochdoerfer commented:
No worries. You can skip the git hook if that is easier for you ;) As long as we get the build green, I am fine.

xaevik commented:
@shochdoerfer it looks as though `phpstan.neon` had an outdated `ignoreErrors` value. I removed it, ran tests locally and squashed all the commits. CI should (hopefully) return green this time.

shochdoerfer commented:
@xaevik looks good! Thx for the help. I know the version upgrades can be a bit annoying. Very much appreciate your support.

Will try to release a new package in the next few days. Just need to find the time for it.

shochdoerfer commented:
@xaevik just wondering: Should we also remove the older PHP versions and Magento module versions from composer.json? I mean older versions of the module still work with those older versions and ideally the CI pipeline covers all versions we actively support.

xaevik commented:
@shochdoerfer my general rule of thumb is to only support what is not considered end-of-life, in this case Magento >= 2.4.4 and PHP >= 8.1 (I need to add Magento 2.4.6 to the CI as I forgot). 

I'll make some further refinements to composer.json as well to bring everything up to snuff.

xaevik commented:
@shochdoerfer all right I've made the following changes:

- Updated `composer.json` dependencies to only support Magento >= 2.4.4 (see [life cycle](https://experienceleague.adobe.com/docs/commerce-operations/release/planning/lifecycle-policy.html?lang=en) for details).
- Bumped the version from `5.1.0` to `6.0.0` as this is considered a breaking change.
- Updated `ci.yml` to test against `2.4.4`, `2.4.5`, and `2.4.6` (tested locally with [nektos/act](https://github.com/nektos/act))

*Technically* the direct PHP requirement in `composer.json` is not necessary as the direct dependency on `magento/framework` already takes care of this, but if you ever want to employ outside tools such as [Mend Renovate](https://www.mend.io/renovate/) or [Dependabot](https://github.com/dependabot) it helps.

xaevik commented:
@shochdoerfer hey no worries, it's out now and now everyone can benefit from the change.

barryvdh commented:
Can you tag an update for this? Thanks!

shochdoerfer commented:
@barryvdh here you go https://github.com/bitExpert/magento2-force-login/releases/tag/v5.2.0

barryvdh commented:
Thanks!
